open_dread_rando/door_patcher.py

```diff
@@ -320,4 +333,59 @@ def update_minimap_for_shield(self, shield: Container, shield_type: ActorData, d
     def remove_all_shields(self):
         for scenario in ["s010_cave", "s020_magma", "s030_baselab", "s040_aqua", "s050_forest", "s060_quarantine", "s070_basesanc", "s080_shipyard", "s090_skybase"]:
             bmmap = self.editor.get_scenario_map(scenario)
-            bmmap.raw.Root.mapBlockages = Container()
\ No newline at end of file
+            bmmap.raw.Root.mapBlockages = Container()
+    
+    def rename_all_shields(self):
+        for scenario in ["s010_cave", "s020_magma", "s030_baselab", "s040_aqua", "s050_forest", "s060_quarantine", "s070_basesanc", "s080_shipyard", "s090_skybase"]:
+            brfld = self.editor.get_scenario(scenario)
+            shielded_doors = []
+            for layer_name, actor_name, actor in list(brfld.all_actors()):
+                if actor_name == "DreadRando_CUDoor":
+                    continue
+                if not is_door(actor):
+                    continue
+
+                if actor.oActorDefLink[9:] not in ActorData.DOOR_POWER.actordefs:
+                    continue
+
+                shielded_doors.append(actor)
+            
+            for door in shielded_doors:
+                self.rename_shields(door, scenario)
+            
+
+    def rename_shields(self, door: Container, scenario: str):
+        life_comp = door.pComponents.LIFE
+        for link_name in ["wpLeftDoorShieldEntity", "wpRightDoorShieldEntity"]:
+            link = life_comp[link_name]
+            if link == "{EMPTY}":
+                continue
+            
+            # get shield actor and cache its sName
+            shieldActor = self.editor.resolve_actor_reference(self.editor.reference_for_link(link, scenario))
+            old_sName = shieldActor.sName
+
+            # skip hdoors
+            if "db_hdoor" in old_sName:
+                continue
+
+            # reclaim old shield id if this is a RandoShield
+            shieldId = int(shieldActor.sName.split("_")[1]) if "RandoShield" in shieldActor.sName else None
```

Comment:
These 3 lines show up identically in `remove_shields`, would be great to have a separate function that does this.